roles/openshift_deploy_router/tasks/main.yml

```diff
@@ -0,0 +1,24 @@
+---
+- name: Deploy router
+  shell: osadm router --create --credentials=/var/lib/openshift/openshift.local.certificates/openshift-router/.kubeconfig --images='registry.access.redhat.com/openshift3_beta/ose-${component}:${version}'
```

Comment:
osadm router supports the --selector flag, which will allow you to specify the node selector without having to do an osc update.